bundles/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/internal/handler/ZmMonitorHandler.java

```diff
@@ -0,0 +1,303 @@
+        updateChannelState(CHANNEL_WEEK_EVENTS, new DecimalType(m.getWeekEvents()));
+        updateChannelState(CHANNEL_MONTH_EVENTS, new DecimalType(m.getMonthEvents()));
+        updateChannelState(CHANNEL_TOTAL_EVENTS, new DecimalType(m.getTotalEvents()));
+        updateChannelState(CHANNEL_IMAGE_URL, new StringType(m.getImageUrl()));
+        updateChannelState(CHANNEL_VIDEO_URL, new StringType(m.getVideoUrl()));
+        updateChannelState(CHANNEL_ALARM, m.isAlarm() ? OnOffType.ON : OnOffType.OFF);
+        updateChannelState(CHANNEL_STATE, new StringType(m.getState().toString()));
+        if (!m.isAlarm()) {
+            updateChannelState(CHANNEL_TRIGGER_ALARM, m.isAlarm() ? OnOffType.ON : OnOffType.OFF);
+        }
+        Event event = m.getLastEvent();
+        if (event == null) {
+            clearEventChannels();
+        } else if (event.getEnd() != null) {
+            // If end is null, assume event hasn't completed yet
+            logger.trace("Monitor {}: Id:{}, Frames:{}, AlarmFrames:{}, Length:{}", m.getId(), event.getId(),
+                    event.getFrames(), event.getAlarmFrames(), event.getLength());
+            updateChannelState(CHANNEL_EVENT_ID, new StringType(event.getId()));
+            updateChannelState(CHANNEL_EVENT_NAME, new StringType(event.getName()));
+            updateChannelState(CHANNEL_EVENT_CAUSE, new StringType(event.getCause()));
+            updateChannelState(CHANNEL_EVENT_NOTES, new StringType(event.getNotes()));
+            updateChannelState(CHANNEL_EVENT_START, new DateTimeType(
+                    ZonedDateTime.ofInstant(event.getStart().toInstant(), timeZoneProvider.getTimeZone())));
+            updateChannelState(CHANNEL_EVENT_END, new DateTimeType(
+                    ZonedDateTime.ofInstant(event.getEnd().toInstant(), timeZoneProvider.getTimeZone())));
+            updateChannelState(CHANNEL_EVENT_FRAMES, new DecimalType(event.getFrames()));
+            updateChannelState(CHANNEL_EVENT_ALARM_FRAMES, new DecimalType(event.getAlarmFrames()));
+            updateChannelState(CHANNEL_EVENT_LENGTH, new QuantityType<Time>(event.getLength(), SmartHomeUnits.SECOND));
+        }
+    }
+
+    private void clearEventChannels() {
+        updateChannelState(CHANNEL_EVENT_ID, UnDefType.NULL);
+        updateChannelState(CHANNEL_EVENT_NAME, UnDefType.NULL);
+        updateChannelState(CHANNEL_EVENT_CAUSE, UnDefType.NULL);
+        updateChannelState(CHANNEL_EVENT_NOTES, UnDefType.NULL);
+        updateChannelState(CHANNEL_EVENT_START, UnDefType.NULL);
+        updateChannelState(CHANNEL_EVENT_END, UnDefType.NULL);
+        updateChannelState(CHANNEL_EVENT_FRAMES, UnDefType.NULL);
+        updateChannelState(CHANNEL_EVENT_ALARM_FRAMES, UnDefType.NULL);
+        updateChannelState(CHANNEL_EVENT_LENGTH, UnDefType.NULL);
+    }
+
+    private void refreshImage() {
+        if (isLinked(CHANNEL_IMAGE)) {
+            scheduler.execute(this::getImage);
```

Comment:
`refreshImage()` is already called asynchronously, why does `getImage()` need to be asynchronous then?

Reply:
My original thinking was that I didn't want `refreshImage` to block waiting for the image. But, I can't think of a reason why that would be necessary. I've removed it.